### DISCUSSION TITLE
Test of hc-sr04 fails

### DISCUSSION THREAD

**Discussion by Frida854** [Moddable-OpenSource/moddable#1070] [Q&A] [answered]
I try 'moddable/examples/drivers/sensors/hc-sr04' but encounter this problem:
'/Users/poulborg/projects-esp32/hc-sr04/main.js (21) # Break: main: get device: undefined variable!'

https://github.com/Moddable-OpenSource/moddable/blob/2a4641e93239b371265bda4b588a7c2e718c67cf/examples/drivers/sensors/hc-sr04/single-sensor/main.js#L21(url)

How do I proceed?

**Comment by andycarle** [comment, marked as answer]
Could you please share what command you're using to build? My guess would be that you're building for a target that does not support ECMA-419 I/O.

**Reply by Frida854** [reply]
mcconfig -d -m -p esp32
![esp32-pico-kit-v4-pinout](https://user-images.githubusercontent.com/18483582/227611184-a271f660-0b82-4896-aa3e-2f70ff6e010a.png)

**Reply by andycarle** [reply]
Gotcha. That will not work for the reason I speculated above--that build target does not support ECMA-419 I/O, which is needed by the example you're trying to run.

The ideal fix would be for someone to add this particular development board (ESP32-PICO-KIT) as a new build target with a proper I/O provider. That requires some thought on what default pins to assign, what built-in functionality the board has, etc. 

I have not done that, but I just put together a generic target that might be sufficient for you to get started: [esp32_pico_kit.zip](https://github.com/Moddable-OpenSource/moddable/files/11066300/esp32_pico_kit.zip). Please try unzipping that folder into `$MODDABLE/build/devices/esp32/targets/` and then building with `mcconfig -d -m -p esp32/esp32_pico_kit`.

Let me know how it goes!

**Comment by Frida854** [comment]
Even after misfitting the sensor, it works now.
Thank you.

**Reply by andycarle** [reply]
Excellent; glad to hear it.
